From fa170d611d192bcc66a826d58c9f2af43d4c5f41 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 2 Aug 2013 12:40:41 +0800
Subject: [PATCH 227/281] ENGR00273616 OTA build failure in all fsl platform

Merge recovery.fstab into fstab.freescale
Define TARGET_RECOVERY_FSTAB based on ext4 or ubifs setting

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 arm2_6dq/BoardConfig.mk             |  2 ++
 arm2_6dq/fstab.freescale            | 11 +++++++----
 arm2_6dq/recovery.fstab             | 13 -------------
 arm2_6sl/BoardConfig.mk             |  2 ++
 arm2_6sl/fstab.freescale            | 11 +++++++----
 arm2_6sl/recovery.fstab             | 13 -------------
 evk_6sl/BoardConfig.mk              | 12 ++++++++++++
 evk_6sl/fstab.freescale             | 11 +++++++----
 evk_6sl/recovery.fstab              | 13 -------------
 hdmidongle_6dq/BoardConfig.mk       |  2 ++
 hdmidongle_6dq/fstab.freescale      | 11 +++++++----
 hdmidongle_6dq/fstab_nand.freescale |  4 +++-
 hdmidongle_6dq/recovery.fstab       | 10 ----------
 hdmidongle_6dq/recovery_nand.fstab  |  7 -------
 imx53_smd/BoardConfig.mk            |  2 ++
 imx53_smd/recovery.fstab            | 10 ----------
 sabreauto_6q/BoardConfig.mk         |  2 ++
 sabreauto_6q/fstab.freescale        | 11 +++++++----
 sabreauto_6q/fstab_nand.freescale   |  6 ++++--
 sabreauto_6q/recovery.fstab         |  8 --------
 sabreauto_6q/recovery_nand.fstab    |  6 ------
 sabresd_6dq/BoardConfig.mk          |  2 ++
 sabresd_6dq/fstab.freescale         | 12 +++++++-----
 sabresd_6dq/recovery.fstab          | 17 -----------------
 24 files changed, 73 insertions(+), 125 deletions(-)
 delete mode 100644 arm2_6dq/recovery.fstab
 delete mode 100644 arm2_6sl/recovery.fstab
 delete mode 100644 evk_6sl/recovery.fstab
 delete mode 100644 hdmidongle_6dq/recovery.fstab
 delete mode 100644 hdmidongle_6dq/recovery_nand.fstab
 delete mode 100644 imx53_smd/recovery.fstab
 delete mode 100644 sabreauto_6q/recovery.fstab
 delete mode 100644 sabreauto_6q/recovery_nand.fstab
 delete mode 100644 sabresd_6dq/recovery.fstab

diff --git a/arm2_6dq/BoardConfig.mk b/arm2_6dq/BoardConfig.mk
index 12f8b54..3b7e567 100755
--- a/arm2_6dq/BoardConfig.mk
+++ b/arm2_6dq/BoardConfig.mk
@@ -12,10 +12,12 @@ BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
 ifeq ($(BUILD_TARGET_FS),ubifs)
+TARGET_RECOVERY_FSTAB = device/fsl/arm2_6dq/fstab_nand.freescale
 # build ubifs for nand devices
 PRODUCT_COPY_FILES +=	\
 	device/fsl/arm2_6dq/fstab_nand.freescale:root/fstab.freescale
 else
+TARGET_RECOVERY_FSTAB = device/fsl/arm2_6dq/fstab.freescale
 # build for ext4
 PRODUCT_COPY_FILES +=	\
 	device/fsl/arm2_6dq/fstab.freescale:root/fstab.freescale
diff --git a/arm2_6dq/fstab.freescale b/arm2_6dq/fstab.freescale
index b81ba34..75e8146 100644
--- a/arm2_6dq/fstab.freescale
+++ b/arm2_6dq/fstab.freescale
@@ -5,8 +5,11 @@
 
 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
-/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
-/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
-/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+/dev/block/mmcblk0p5    /system      ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data        ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache       ext4    nosuid,nodev,nomblk_io_submit                                                    wait
+/dev/block/mmcblk0p7    /device      ext4    ro,nosuid,nodev                                                                  wait
+/dev/block/mmcblk0p1    /boot        emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p2    /recovery    emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p8    /misc        emmc    defaults                                                                         defaults
 
diff --git a/arm2_6dq/recovery.fstab b/arm2_6dq/recovery.fstab
deleted file mode 100644
index ef1b961..0000000
--- a/arm2_6dq/recovery.fstab
+++ /dev/null
@@ -1,13 +0,0 @@
-/boot emmc /dev/block/mmcblk0p1
-/recovery emmc /dev/block/mmcblk0p2
-/system ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-/data ext4 /dev/block/mmcblk0p4 reserved=32768
-/misc  emmc /dev/block/mmcblk0p8
-
-# UBI support is under below:
-#/system ubifs /dev/ubi1_0
-#/data ubifs /dev/ubi1_2
-#/cache ubifs /dev/ubi1_1
-
-/sdcard vfat /dev/block/mmcblk1p1
diff --git a/arm2_6sl/BoardConfig.mk b/arm2_6sl/BoardConfig.mk
index 5029c9d..26aa6e2 100755
--- a/arm2_6sl/BoardConfig.mk
+++ b/arm2_6sl/BoardConfig.mk
@@ -12,10 +12,12 @@ BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
 ifeq ($(BUILD_TARGET_FS),ubifs)
+TARGET_RECOVERY_FSTAB = device/fsl/arm2_6sl/fstab_nand.freescale
 # build ubifs for nand devices
 PRODUCT_COPY_FILES +=	\
 	device/fsl/arm2_6sl/fstab_nand.freescale:root/fstab.freescale
 else
+TARGET_RECOVERY_FSTAB = device/fsl/arm2_6sl/fstab.freescale
 # build for ext4
 PRODUCT_COPY_FILES +=	\
 	device/fsl/arm2_6sl/fstab.freescale:root/fstab.freescale
diff --git a/arm2_6sl/fstab.freescale b/arm2_6sl/fstab.freescale
index dcbcd14..68a87e6 100644
--- a/arm2_6sl/fstab.freescale
+++ b/arm2_6sl/fstab.freescale
@@ -5,8 +5,11 @@
 
 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
-/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
-/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
-/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+/dev/block/mmcblk0p5    /system      ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data        ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache       ext4    nosuid,nodev,nomblk_io_submit                                                    wait
+/dev/block/mmcblk0p7    /device      ext4    ro,nosuid,nodev                                                                  wait
+/dev/block/mmcblk0p1    /boot        emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p2    /recovery    emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p8    /misc        emmc    defaults                                                                         defaults
 
diff --git a/arm2_6sl/recovery.fstab b/arm2_6sl/recovery.fstab
deleted file mode 100644
index 9f9586c..0000000
--- a/arm2_6sl/recovery.fstab
+++ /dev/null
@@ -1,13 +0,0 @@
-/boot   emmc /dev/block/mmcblk0p1
-/recovery emmc /dev/block/mmcblk0p2
-/system ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-/data ext4 /dev/block/mmcblk0p4 reserved=32768
-/misc  emmc  /dev/block/mmcblk0p8
-
-# UBI support is under below:
-#/system ubifs /dev/ubi1_0
-#/data ubifs /dev/ubi1_2
-#/cache ubifs /dev/ubi1_1
-
-/sdcard vfat /dev/block/mmcblk1p1
diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index 23da170..0a30106 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -10,6 +10,18 @@ include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
 BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
+ifeq ($(BUILD_TARGET_FS),ubifs)
+TARGET_RECOVERY_FSTAB = device/fsl/evk_6sl/fstab_nand.freescale
+# build ubifs for nand devices
+PRODUCT_COPY_FILES +=	\
+	device/fsl/evk_6sl/fstab_nand.freescale:root/fstab.freescale
+else
+TARGET_RECOVERY_FSTAB = device/fsl/evk_6sl/fstab.freescale
+# build for ext4
+PRODUCT_COPY_FILES +=	\
+	device/fsl/evk_6sl/fstab.freescale:root/fstab.freescale
+endif # BUILD_TARGET_FS
+
 TARGET_BOOTLOADER_BOARD_NAME := EVK
 PRODUCT_MODEL := EVK_MX6SL
 
diff --git a/evk_6sl/fstab.freescale b/evk_6sl/fstab.freescale
index 0d20f60..a6fca58 100644
--- a/evk_6sl/fstab.freescale
+++ b/evk_6sl/fstab.freescale
@@ -5,8 +5,11 @@
 
 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
-/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
-/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
-/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+/dev/block/mmcblk0p5    /system      ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data        ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache       ext4    nosuid,nodev,nomblk_io_submit                                                    wait
+/dev/block/mmcblk0p7    /device      ext4    ro,nosuid,nodev                                                                  wait
+/dev/block/mmcblk0p1    /boot        emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p2    /recovery    emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p8    /misc        emmc    defaults                                                                         defaults
 
diff --git a/evk_6sl/recovery.fstab b/evk_6sl/recovery.fstab
deleted file mode 100644
index 9f9586c..0000000
--- a/evk_6sl/recovery.fstab
+++ /dev/null
@@ -1,13 +0,0 @@
-/boot   emmc /dev/block/mmcblk0p1
-/recovery emmc /dev/block/mmcblk0p2
-/system ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-/data ext4 /dev/block/mmcblk0p4 reserved=32768
-/misc  emmc  /dev/block/mmcblk0p8
-
-# UBI support is under below:
-#/system ubifs /dev/ubi1_0
-#/data ubifs /dev/ubi1_2
-#/cache ubifs /dev/ubi1_1
-
-/sdcard vfat /dev/block/mmcblk1p1
diff --git a/hdmidongle_6dq/BoardConfig.mk b/hdmidongle_6dq/BoardConfig.mk
index 12aa8d9..c7f95dc 100755
--- a/hdmidongle_6dq/BoardConfig.mk
+++ b/hdmidongle_6dq/BoardConfig.mk
@@ -11,10 +11,12 @@ BUILD_TARGET_FS ?= ubifs
 include device/fsl/imx6/imx6_target_fs.mk
 
 ifeq ($(BUILD_TARGET_FS),ubifs)
+TARGET_RECOVERY_FSTAB = device/fsl/hdmidongle_6dq/fstab_nand.freescale
 # build ubifs for nand devices
 PRODUCT_COPY_FILES +=	\
 	device/fsl/hdmidongle_6dq/fstab_nand.freescale:root/fstab.freescale
 else
+TARGET_RECOVERY_FSTAB = device/fsl/hdmidongle_6dq/fstab.freescale
 # build for ext4
 PRODUCT_COPY_FILES +=	\
 	device/fsl/hdmidongle_6dq/fstab.freescale:root/fstab.freescale
diff --git a/hdmidongle_6dq/fstab.freescale b/hdmidongle_6dq/fstab.freescale
index 65e1db6..89dc642 100644
--- a/hdmidongle_6dq/fstab.freescale
+++ b/hdmidongle_6dq/fstab.freescale
@@ -5,8 +5,11 @@
 
 /devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
-/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
-/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
-/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+/dev/block/mmcblk0p5    /system      ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data        ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache       ext4    nosuid,nodev,nomblk_io_submit                                                    wait
+/dev/block/mmcblk0p7    /device      ext4    ro,nosuid,nodev                                                                  wait
+/dev/block/mmcblk0p1    /boot        emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p2    /recovery    emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p8    /misc        emmc    defaults                                                                         defaults
 
diff --git a/hdmidongle_6dq/fstab_nand.freescale b/hdmidongle_6dq/fstab_nand.freescale
index 99a0ed4..4e22c5a 100644
--- a/hdmidongle_6dq/fstab_nand.freescale
+++ b/hdmidongle_6dq/fstab_nand.freescale
@@ -10,4 +10,6 @@
 ubi0:system	/system		ubifs 	ro					wait
 ubi0:data       /data           ubifs   nosuid,nodev,nodiratime,noatime         wait
 ubi0:cache      /cache          ubifs   nosuid,nodev                            wait
-ubi0:device     /device         ubifs   nosuid,nodev                         wait
+ubi0:device     /device         ubifs   nosuid,nodev                            wait
+/dev/mtd/mtd1   /boot           mtd     defaults                                defaults
+/dev/mtd/mtd2   /recovery       mtd     defaults                                defaults
diff --git a/hdmidongle_6dq/recovery.fstab b/hdmidongle_6dq/recovery.fstab
deleted file mode 100644
index b2e07c9..0000000
--- a/hdmidongle_6dq/recovery.fstab
+++ /dev/null
@@ -1,10 +0,0 @@
-# Use below lines if you use eMMC:
-/sdcard vfat /dev/block/mmcblk0p1
-/boot emmc /dev/block/mmcblk0p1
-/recovery emmc /dev/block/mmcblk0p2
-/system ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-/data ext4 /dev/block/mmcblk0p4 reserved=32768
-/misc  emmc /dev/block/mmcblk0p8
-
-/sdcard vfat /dev/block/mmcblk1p1
diff --git a/hdmidongle_6dq/recovery_nand.fstab b/hdmidongle_6dq/recovery_nand.fstab
deleted file mode 100644
index 0e9e3ca..0000000
--- a/hdmidongle_6dq/recovery_nand.fstab
+++ /dev/null
@@ -1,7 +0,0 @@
-# Use below lines if you use NAND:
-/boot mtd /dev/mtd/mtd1
-/recovery mtd /dev/mtd/mtd2
-/system ubifs /dev/ubi0_0
-/data ubifs /dev/ubi0_2
-/cache ubifs /dev/ubi0_1
-/sdcard vfat /dev/block/mmcblk0p1
diff --git a/imx53_smd/BoardConfig.mk b/imx53_smd/BoardConfig.mk
index cca39d2..3c882fb 100755
--- a/imx53_smd/BoardConfig.mk
+++ b/imx53_smd/BoardConfig.mk
@@ -10,6 +10,8 @@ BOARD_SOC_CLASS := IMX5X
 BOARD_SOC_TYPE := IMX53
 PRODUCT_MODEL := IMX53-SMD
 
+TARGET_RECOVERY_FSTAB = device/fsl/imx53_smd/fstab.freescale
+
 # UNITE is a virtual device support both atheros and realtek wifi(ar6103 and rtl8723as)
 BOARD_WLAN_DEVICE            := UNITE
 WPA_SUPPLICANT_VERSION       := VER_0_8_UNITE
diff --git a/imx53_smd/recovery.fstab b/imx53_smd/recovery.fstab
deleted file mode 100644
index a0153c7..0000000
--- a/imx53_smd/recovery.fstab
+++ /dev/null
@@ -1,10 +0,0 @@
-/system ext4 /dev/block/mmcblk0p2
-/data ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-
-# UBI support is under below:
-#/system ubifs /dev/ubi1_0
-#/data ubifs /dev/ubi1_2
-#/cache ubifs /dev/ubi1_1
-
-/sdcard vfat /dev/block/mmcblk1p1
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index 18a2fa2..3fd6f0b 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -11,10 +11,12 @@ BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
 ifeq ($(BUILD_TARGET_FS),ubifs)
+TARGET_RECOVERY_FSTAB = device/fsl/sabreauto_6q/fstab_nand.freescale
 # build ubifs for nand devices
 PRODUCT_COPY_FILES +=	\
 	device/fsl/sabreauto_6q/fstab_nand.freescale:root/fstab.freescale
 else
+TARGET_RECOVERY_FSTAB = device/fsl/sabreauto_6q/fstab.freescale
 # build for ext4
 PRODUCT_COPY_FILES +=	\
 	device/fsl/sabreauto_6q/fstab.freescale:root/fstab.freescale
diff --git a/sabreauto_6q/fstab.freescale b/sabreauto_6q/fstab.freescale
index 0d20f60..a6fca58 100644
--- a/sabreauto_6q/fstab.freescale
+++ b/sabreauto_6q/fstab.freescale
@@ -5,8 +5,11 @@
 
 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
-/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
-/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
-/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+/dev/block/mmcblk0p5    /system      ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data        ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache       ext4    nosuid,nodev,nomblk_io_submit                                                    wait
+/dev/block/mmcblk0p7    /device      ext4    ro,nosuid,nodev                                                                  wait
+/dev/block/mmcblk0p1    /boot        emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p2    /recovery    emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p8    /misc        emmc    defaults                                                                         defaults
 
diff --git a/sabreauto_6q/fstab_nand.freescale b/sabreauto_6q/fstab_nand.freescale
index 3d928bb..24d20ba 100644
--- a/sabreauto_6q/fstab_nand.freescale
+++ b/sabreauto_6q/fstab_nand.freescale
@@ -7,7 +7,9 @@
 
 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-ubi0:system	/system		ubifs 	ro					wait
+ubi0:system	    /system		    ubifs 	ro                                      wait
 ubi0:data       /data           ubifs   nosuid,nodev,nodiratime,noatime         wait
 ubi0:cache      /cache          ubifs   nosuid,nodev                            wait
-ubi0:device     /device         ubifs   nosuid,nodev                         wait
+ubi0:device     /device         ubifs   nosuid,nodev                            wait
+/dev/mtd/mtd1   /boot           mtd     defaults                                defaults
+/dev/mtd/mtd2   /recovery       mtd     defaults                                defaults
diff --git a/sabreauto_6q/recovery.fstab b/sabreauto_6q/recovery.fstab
deleted file mode 100644
index a6e5f10..0000000
--- a/sabreauto_6q/recovery.fstab
+++ /dev/null
@@ -1,8 +0,0 @@
-/boot emmc /dev/block/mmcblk0p1
-/recovery emmc /dev/block/mmcblk0p2
-/system ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-/data ext4 /dev/block/mmcblk0p4 reserved=32768
-/misc  emmc /dev/block/mmcblk0p8
-
-/sdcard vfat /dev/block/mmcblk1p1
diff --git a/sabreauto_6q/recovery_nand.fstab b/sabreauto_6q/recovery_nand.fstab
deleted file mode 100644
index 270041c..0000000
--- a/sabreauto_6q/recovery_nand.fstab
+++ /dev/null
@@ -1,6 +0,0 @@
-# UBI support is under below:
-/boot mtd /dev/mtd/mtd1
-/recovery mtd /dev/mtd/mtd2
-/system ubifs /dev/ubi0_0
-/data ubifs /dev/ubi0_3
-/cache ubifs /dev/ubi0_1
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index a3f315c..100be6c 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -11,10 +11,12 @@ BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
 ifeq ($(BUILD_TARGET_FS),ubifs)
+TARGET_RECOVERY_FSTAB = device/fsl/sabresd_6dq/fstab_nand.freescale
 # build ubifs for nand devices
 PRODUCT_COPY_FILES +=	\
 	device/fsl/sabresd_6dq/fstab_nand.freescale:root/fstab.freescale
 else
+TARGET_RECOVERY_FSTAB = device/fsl/sabresd_6dq/fstab.freescale
 # build for ext4
 PRODUCT_COPY_FILES +=	\
 	device/fsl/sabresd_6dq/fstab.freescale:root/fstab.freescale
diff --git a/sabresd_6dq/fstab.freescale b/sabresd_6dq/fstab.freescale
index dcbcd14..c2b737a 100644
--- a/sabresd_6dq/fstab.freescale
+++ b/sabresd_6dq/fstab.freescale
@@ -5,8 +5,10 @@
 
 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
 /devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
-/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
-/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
-/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
-/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
-
+/dev/block/mmcblk0p5    /system      ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data        ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache       ext4    nosuid,nodev,nomblk_io_submit                                                    wait
+/dev/block/mmcblk0p7    /device      ext4    ro,nosuid,nodev                                                                  wait
+/dev/block/mmcblk0p1    /boot        emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p2    /recovery    emmc    defaults                                                                         defaults
+/dev/block/mmcblk0p8    /misc        emmc    defaults                                                                         defaults
diff --git a/sabresd_6dq/recovery.fstab b/sabresd_6dq/recovery.fstab
deleted file mode 100644
index 84e242b..0000000
--- a/sabresd_6dq/recovery.fstab
+++ /dev/null
@@ -1,17 +0,0 @@
-/boot   emmc /dev/block/mmcblk0p1
-/recovery emmc /dev/block/mmcblk0p2
-/system ext4 /dev/block/mmcblk0p5
-/cache ext4 /dev/block/mmcblk0p6
-/data ext4 /dev/block/mmcblk0p4 reserved=32768
-/misc  emmc  /dev/block/mmcblk0p8
-/sdcard vfat /dev/block/mmcblk1p1
-
-# Use below lines if you use NAND:
-#/boot mtd /dev/mtd/mtd1
-#/recovery mtd /dev/mtd/mtd2
-#/system ubifs /dev/ubi0_0
-#/data ubifs /dev/ubi0_2
-#/cache ubifs /dev/ubi0_1
-#/sdcard vfat /dev/block/mmcblk0p1
-
-
-- 
1.8.0

